pl_bolts/callbacks/vision/confused_logit.py

```diff
@@ -1,14 +1,17 @@
 import importlib
+from typing import Sequence
 
 import torch
-from pytorch_lightning import Callback
-from torch import nn
+from pytorch_lightning import Callback, LightningModule, Trainer
+from torch import Tensor, nn
 
 from pl_bolts.utils.warnings import warn_missing_pkg
 
-_MATPLOTLIB_AVAILABLE = importlib.util.find_spec("matplotlib") is not None
+_MATPLOTLIB_AVAILABLE = importlib.util.find_spec("matplotlib") is not None  # type: ignore[attr-defined]
```

Comment:
We actually have similar variables for other optional packages in `pl_bolts/utils/__init__.py`:
https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/a72766c461a498ab67b7c2efe7e0d49224290837/pl_bolts/utils/__init__.py#L6-L10
So, could you move the definition of `_MATPLOTLIB_AVAILABLE` to the linked file and import the variable here?